test/e2e/user_management_test.go

```diff
@@ -128,6 +128,55 @@ func (s *userManagementTestSuite) TestUserDeactivation() {
 		})
 	})
 
+	s.T().Run("tests for basic tier with deactivation disabled", func(t *testing.T) {
+		// Let's create a tier with deactivation disabled
+		noDeactivationTier := CreateNSTemplateTier(t, s.ctx, s.hostAwait, "deactivation-tier")
```

Comment:
But why? You just added a tier `basic-no-deactivation` that has the deactivation disabled - you don't have to create a new one.

Reply:
Somewhat related to the comment above, I figured we wanted to test the general functionality and not have it dependent on the existence of any tiers?

Reply:
Well... since it's e2e tests the main goal of these tests is the actual functionality available to the users. We don't generally care about the implementation details.
Basically some test which verifies what happens to the user who is provisioned with the basic-no-deactivation tier (the user should be deactivated) is more valuable than testing an abstract deactivation mechanism of some non-existing tier.
It's not always possible though and sometimes we have to do some tricks like copying tiers. But those tricks are just workarounds of technical limitations we face in the e2e environment.

It's different in unit tests.

Reply:
Sounds good. I've updated the test to just use the real tiers `basic` and `basicdeactivationdisabled`

Reply:
I agree with @alexeykazakov 
In this care, we need to verify the actual configuration of the Tier, not the general deactivation logic.
In addition, we can easily use the tier without creating a new one - in other words, the workaround mentioned by Alexey is not needed